go/test/endtoend/onlineddl/flow/onlineddl_flow_test.go

```diff
@@ -0,0 +1,664 @@
+		})
+		t.Run("init table", func(t *testing.T) {
+			// Populates table. Makes work for vcopier.
+			initTable(t)
+		})
+		t.Run("migrate", func(t *testing.T) {
+			ctx, cancel := context.WithCancel(ctx)
+			defer cancel()
+
+			workloadCtx, cancelWorkload := context.WithCancel(ctx)
+			defer cancelWorkload()
+
+			t.Run("routine throttler check", func(t *testing.T) {
+				go func() {
+					ticker := time.NewTicker(500 * time.Millisecond)
+					defer ticker.Stop()
+					for {
+						_, statusCode, err := throttlerCheck(primaryTablet.VttabletProcess, throttlerapp.OnlineDDLName)
+						assert.NoError(t, err)
+						throttleWorkload.Store(statusCode != http.StatusOK)
+						select {
+						case <-ticker.C:
+						case <-workloadCtx.Done():
+							t.Logf("Terminating routine throttler check")
+							return
+						}
+					}
+				}()
+			})
+
+			var wg sync.WaitGroup
+			t.Run("generate workload", func(t *testing.T) {
+				// Create work for vplayer.
+				// This workload will consider throttling state and avoid generating DMLs if throttled.
+				wg.Add(1)
+				go func() {
+					defer cancel()
+					defer t.Logf("Terminating workload")
+					defer wg.Done()
+					runMultipleConnections(workloadCtx, t)
+				}()
+			})
+			appliedDMLStart := totalAppliedDML.Load()
+
+			hint := "post_completion_hint"
+			var uuid string
+			t.Run("submit migration", func(t *testing.T) {
+				uuid = testOnlineDDLStatement(t, fmt.Sprintf(alterHintStatement, hint), "online --postpone-completion", "", true)
+			})
+			t.Run("wait for ready_to_complete", func(t *testing.T) {
+				waitForReadyToComplete(t, uuid, true)
+			})
+			t.Run("validating running status", func(t *testing.T) {
+				onlineddl.CheckMigrationStatus(t, &vtParams, shards, uuid, schema.OnlineDDLStatusRunning)
+			})
+			t.Run("throttle online-ddl", func(t *testing.T) {
+				onlineddl.ThrottleAllMigrations(t, &vtParams)
+				onlineddl.CheckThrottledApps(t, &vtParams, throttlerapp.OnlineDDLName, true)
+
+				for _, tab := range tablets {
+					body, err := throttleApp(tab.VttabletProcess, throttlerapp.OnlineDDLName)
+					assert.NoError(t, err)
+					assert.Contains(t, body, throttlerapp.OnlineDDLName)
+				}
+				waitForThrottleCheckStatus(t, throttlerapp.OnlineDDLName, primaryTablet, http.StatusExpectationFailed)
+			})
+			t.Run("unthrottle online-ddl", func(t *testing.T) {
+				onlineddl.UnthrottleAllMigrations(t, &vtParams)
+				onlineddl.CheckThrottledApps(t, &vtParams, throttlerapp.OnlineDDLName, false)
+
+				for _, tab := range tablets {
+					body, err := unthrottleApp(tab.VttabletProcess, throttlerapp.OnlineDDLName)
+					assert.NoError(t, err)
+					assert.Contains(t, body, throttlerapp.OnlineDDLName)
+				}
+				waitForThrottleCheckStatus(t, throttlerapp.OnlineDDLName, primaryTablet, http.StatusOK)
+			})
+			t.Run("additional wait", func(t *testing.T) {
+				select {
+				case <-time.After(3 * time.Second):
+				case <-ctx.Done():
+					require.Fail(t, "context cancelled")
+				}
+			})
```

Comment:
Waiting just so that we generate more DMLs, and give migration/vreplication more "opportunities" to throttle or to make progress. Adding clarification.

Reply:
Added.